ledger/store/trackerdb/sqlitedriver/sql.go

```diff
@@ -421,6 +426,71 @@ func (qs *accountsDbQueries) LookupAllResources(addr basics.Address) (data []tra
 	return
 }
 
+func (qs *accountsDbQueries) LookupLimitedResources(addr basics.Address, minIdx basics.CreatableIndex, maxCreatables uint64, ctype basics.CreatableType) (data []trackerdb.PersistedResourcesDataWithCreator, rnd basics.Round, err error) {
+	err = db.Retry(func() error {
+		// Query for all resources
+		rows, err0 := qs.lookupLimitedResourcesStmt.Query(addr[:], ctype, minIdx, maxCreatables)
+		if err0 != nil {
+			return err0
+		}
+		defer rows.Close()
+
+		var addrid, aidx sql.NullInt64
+		var dbRound basics.Round
+		data = nil
+		var actAssetBuf []byte
+		var crtAssetBuf []byte
+		var creatorAddrBuf []byte
+		for rows.Next() {
+			err = rows.Scan(&addrid, &dbRound, &aidx, &creatorAddrBuf, &actAssetBuf, &crtAssetBuf)
+			if err != nil {
+				return err
+			}
+			if !addrid.Valid || !aidx.Valid {
+				// we received an entry without any index. This would happen only on the first entry when there are no resources for this address.
+				// ensure this is the first entry, set the round and return
+				if len(data) != 0 {
+					return fmt.Errorf("LookupLimitedResources: unexpected invalid result on non-first resource record: (%v, %v)", addrid.Valid, aidx.Valid)
+				}
+				rnd = dbRound
+				break
+			}
+			var actResData trackerdb.ResourcesData
+			var crtResData trackerdb.ResourcesData
+			err = protocol.Decode(actAssetBuf, &actResData)
+			if err != nil {
+				return err
+			}
+			err = protocol.Decode(crtAssetBuf, &crtResData)
+			if err != nil {
+				return err
+			}
+
+			// Update asset holding data in the creator resource data
+			crtResData.Amount = actResData.Amount
+			crtResData.Freeze = actResData.Freeze
+			crtResData.ResourceFlags = actResData.ResourceFlags
```

Comment:
Because the creator is not necessarily the asset holder (and more often than not isn't). It's called crtResData as that was the result from the database, but after setting these fields it becomes the data returned for the asset holder.